service-node/src/main/scala/org/enterprisedlt/fabric/service/node/process/DockerManagedBox.scala

```diff
@@ -80,8 +81,50 @@ class DockerManagedBox(
 
 
     //=========================================================================
+
+    def pullImageIfNeeded(image: String, forcePull: Boolean = false): Unit = {
+        val images = docker
+          .listImagesCmd()
+          .exec()
+          .iterator()
+        var imageExist = false
+        while (images.hasNext && !imageExist) {
+            if (Option(images.next().getRepoTags).exists(_.contains(image))) {
+                imageExist = true
+            }
+        }
+        if (!imageExist || forcePull) {
+            logger.info(s"Unable to find image $image locally or docker pull has been forced")
+            try {
+                val pullImageCmd = image.split(":") match {
+                    case Array(imageName, imageTag) =>
+                        logger.info(s"Pulling image $imageName with tag $imageTag...")
+                        docker
+                          .pullImageCmd(imageName)
+                          .withTag(imageTag)
+                    case Array(imageName) =>
+                        logger.info(s"Pulling image $imageName with tag latest...")
+                        docker
+                          .pullImageCmd(imageName)
+                          .withTag("latest")
+                }
+                pullImageCmd
+                  .exec(new PullImageResultCallback())
+                  .awaitCompletion(1, TimeUnit.MINUTES)
+            } catch {
+                case e: Exception => logger.error(s"Exception pulling image: $e")
```

Comment:
exception ignored -  make this method to return Either and add appropriate handling

Reply:
Done